src/overlays/actors/ovl_En_Elfbub/z_en_elfbub.c

```diff
@@ -22,27 +30,138 @@ const ActorInit En_Elfbub_InitVars = {
     (ActorFunc)EnElfbub_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80ACE270 = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_PLAYER, OC2_TYPE_1, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_PLAYER,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
     { 16, 32, 0, { 0, 0, 0 } },
 };
 
-#endif
+extern Gfx D_06001000[];
+
+void EnElfbub_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnElfbub* this = THIS;
+    Actor* childActor;
+
+    if (Flags_GetSwitch(globalCtx, ENELFBUB_GET_SWITCHFLAG(&this->actor))) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    ActorShape_Init(&this->actor.shape, 16.0f, func_800B3FC0, 0.2f);
+    this->actor.hintId = 0x16;
+    Actor_SetScale(&this->actor, 1.25f);
+
+    this->actionFunc = EnElfbub_Idle;
+    this->zRot = randPlusMinusPoint5Scaled(65536.0f);
+    this->zRotDelta = 1000;
+    this->xScale = 0.08f;
+
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+
+    childActor = Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_ELFORG,
+                                    this->actor.world.pos.x, this->actor.world.pos.y + 12.0f, this->actor.world.pos.z,
+                                    this->actor.world.rot.x, this->actor.world.rot.y, this->actor.world.rot.z,
+                                    ((ENELFBUB_GET_SWITCHFLAG(&this->actor) & 0x7F) << 9) | 2);
+    if (childActor) {
+        childActor->parent = &this->actor;
+    }
+
+    this->oscillationAngle = 0;
+    this->actor.flags &= ~1;
+}
+
+void EnElfbub_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnElfbub* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnElfbub_Pop(EnElfbub* this, GlobalContext* globalCtx) {
+    static Color_RGBA8 sPrimColor = { 255, 255, 255, 255 };
+    static Color_RGBA8 sEnvColor = { 150, 150, 150, 0 };
+    static Vec3f sAccel = { 0.0f, -0.5f, 0.0f };
+    s32 i;
+    Vec3f velocity;
+    Vec3f pos;
+
+    Math_SmoothStepToF(&this->xyScale, 3.0f, 0.1f, 1000.0f, 0.0f);
+    Math_SmoothStepToF(&this->xScale, 0.2f, 0.1f, 1000.0f, 0.0f);
+    this->zRotDelta += 1000;
+    this->zRot += this->zRotDelta;
+    this->popTimer -= 1;
```

Comment:
You can use `--`. It isn't mandatory tho
```suggestion
    this->popTimer--;
```

Reply:
Remnant of trying little changes to get things to match. Thanks for catching.